### PR TITLE
Cook, Serve, Delicious!

### PR DESCRIPTION
**Pull request by Fraxinus88**
# New Port for Cook, Serve, Delicious!

## Game Information
- **Title**: Cook, Serve, Delicious!
- **URL**: https://store.steampowered.com/app/247020/Cook_Serve_Delicious/

### CFW Tests
[X] AmberELEC
[X] ArkOS
[X] JELOS/ROCKNIX
[X] MuOS

Architecture
[X] armHF
[X] Panfrost (Mainline)

Resolution Tests
Test all major resolutions:
[X] 480x320
[X] 640x480
[X] 720x720 (RGB30)
[X] Higher resolutions (e.g., 1280x720)

## File Structure
- Your port should have the following structure:
  - portname/
    - port.json
    - README.md
    - screenshot.jpg
    - cover.jpg
    - gameinfo.xml
    - Port Name.sh
    - portname/
      - <portfiles here>

## Additional Resources
For an in-depth guide on creating a pull request, refer to: [PortMaster Game Packaging Guide](https://portmaster.games/packaging.html#creating-a-pull-request)

### CODE REVIEW

**Comment by Fraxinus88** [comment]
Kindly help me check with libs can be deleted from cookservedelicious/libs

File: ports/cookservedelicious/Cook, Serve, Delicious!.sh [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+# Display loading splash
+if [ -f "$GAMEDIR/gamedata/game.droid" ]; then
+    $ESUDO ./libs/splash "splash.png" 1 
+    $ESUDO ./libs/splash "splash.png" 2000
+fi
+
+# Run the game
+$GPTOKEYB "gmloadernext" -c "./cookservedelicious.gptk" &
+pm_platform_helper "$GAMEDIR/gmloadernext"
+./gmloadernext game.apk
```
Comment (Cebion) [inline ports/cookservedelicious/Cook, Serve, Delicious!.sh:89]: Does gmloadernext take an argument? I thought this was gmloader only?

File: ports/cookservedelicious/port.json [hunk 2]
```diff
@@ -0,0 +1,33 @@
+{
+  "version": 3,
+  "name": "cookservedelicious.zip",
+  "items": [
+    "Cook",
```
Comment (Cebion) [inline ports/cookservedelicious/port.json:5]: What happened here?

**Comment by JanTrueno** [comment]
Pretty sure you only need Crypto, OpenAL and Zip.

**Review by Cebion — CHANGES_REQUESTED** [review]

File: ports/cookservedelicious/Cook, Serve, Delicious!.sh (hunk 1 above)
Comment (Fraxinus88) [reply ports/cookservedelicious/Cook, Serve, Delicious!.sh:89]: Not too sure, copied over from UFO 50 :p

File: ports/cookservedelicious/Cook, Serve, Delicious!.sh [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+# Variables
+GAMEDIR="/$directory/ports/cookservedelicious"
+
+# CD and set permissions
+cd $GAMEDIR
+> "$GAMEDIR/log.txt" && exec > >(tee "$GAMEDIR/log.txt") 2>&1
+$ESUDO chmod +x -R $GAMEDIR/*
+
+# Exports
+export LD_LIBRARY_PATH="/usr/lib:$GAMEDIR/lib:$GAMEDIR/libs:$LD_LIBRARY_PATH"
```
Comment (kloptops) [inline ports/cookservedelicious/Cook, Serve, Delicious!.sh:28]:
```suggestion
export LD_LIBRARY_PATH="/usr/lib:$GAMEDIR/lib:$LD_LIBRARY_PATH"
```

**Review by kloptops — CHANGES_REQUESTED** [review]

File: ports/cookservedelicious/Cook, Serve, Delicious!.sh (hunk 3 above)
Comment (Fraxinus88) [reply ports/cookservedelicious/Cook, Serve, Delicious!.sh:28]: But there are libs....?

Comment (JanTrueno) [reply ports/cookservedelicious/Cook, Serve, Delicious!.sh:28]: Needs to be this: export LD_LIBRARY_PATH="/usr/lib:$GAMEDIR/libs:$LD_LIBRARY_PATH"

Comment (JanTrueno) [reply ports/cookservedelicious/Cook, Serve, Delicious!.sh:28]: You only need export LD_LIBRARY_PATH="/usr/lib:$GAMEDIR/libs:$LD_LIBRARY_PATH" . GAMEDIR/lib is what gmloadernext needs for itself. So Kloptops is almost right :)

Comment (Fraxinus88) [reply ports/cookservedelicious/Cook, Serve, Delicious!.sh:28]: Oh then i can delete libs after that? let me try

Comment (Fraxinus88) [reply ports/cookservedelicious/Cook, Serve, Delicious!.sh]: okies, updated

File: ports/cookservedelicious/Cook, Serve, Delicious!.sh [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+           "./gamedata/"*.dll \
+	   "./gamedata/Place game files here" 
+    echo "Extra game files removed"
+else    
+    echo "No extra game files to remove"
+fi
+
+# Display loading splash
+if [ -f "$GAMEDIR/gamedata/game.droid" ]; then
+    $ESUDO ./lib/splash "splash.png" 1
```
Comment (cdeletre) [inline ports/cookservedelicious/Cook, Serve, Delicious!.sh:82]:
```suggestion
[ "$CFW_NAME" == "muOS" ] && $ESUDO ./lib/splash "splash.png" 1 # muOS only workaround
```

**Review by cdeletre — CHANGES_REQUESTED** [review]

File: ports/cookservedelicious/Cook, Serve, Delicious!.sh [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+# Display loading splash
+if [ -f "$GAMEDIR/gamedata/game.droid" ]; then
+    $ESUDO ./lib/splash "splash.png" 1 
+    $ESUDO ./lib/splash "splash.png" 2000
+fi
+
+# Run the game
+$GPTOKEYB "gmloadernext" -c "./cookservedelicious.gptk" &
+pm_platform_helper "$GAMEDIR/gmloadernext"
+./gmloadernext game.apk
```
Comment (cdeletre) [inline ports/cookservedelicious/Cook, Serve, Delicious!.sh:89]: `game.apk` argument can be removed (I've check on UFO 50 port and it still load the game)
```suggestion
./gmloadernext
```

**Review by cdeletre — CHANGES_REQUESTED** [review]

**Comment by Fraxinus88** [comment]
Okies, committed :)

**Review by cdeletre — APPROVED** [review]

**Review by Cebion — APPROVED** [review]

**Pull request merged**